internal/msalbase/token_response.go

```diff
@@ -22,76 +22,78 @@ type tokenResponseJSONPayload struct {
 	Scope        string `json:"scope"`
 	IDToken      string `json:"id_token"`
 	ClientInfo   string `json:"client_info"`
+
+	AdditionalFields map[string]interface{}
 }
 
-//ClientInfoJSONPayload is used to create a Home Account ID for an account
+// ClientInfoJSONPayload is used to create a Home Account ID for an account.
 type ClientInfoJSONPayload struct {
 	UID  string `json:"uid"`
 	Utid string `json:"utid"`
 }
 
-//TokenResponse is the information that is returned from a token endpoint during a token acquisition flow
+// TokenResponse is the information that is returned from a token endpoint during a token acquisition flow.
 type TokenResponse struct {
-	baseResponse   *OAuthResponseBase
+	baseResponse   OAuthResponseBase
 	AccessToken    string
 	RefreshToken   string
-	IDToken        *IDToken
+	IDToken        IDToken
 	FamilyID       string
 	GrantedScopes  []string
 	declinedScopes []string
 	ExpiresOn      time.Time
 	ExtExpiresOn   time.Time
 	rawClientInfo  string
-	ClientInfo     *ClientInfoJSONPayload
+	ClientInfo     ClientInfoJSONPayload
 }
 
-//HasAccessToken checks if the TokenResponse has an access token secret
-func (tr *TokenResponse) HasAccessToken() bool {
+// HasAccessToken checks if the TokenResponse has an access token secret.
+func (tr TokenResponse) HasAccessToken() bool {
```

Comment:
nit: I would remove "secret" from this line and line 55, that doesn't make sense. not sure why it was there.

Reply:
"secret" is how we store the actual token in the cache, I understood the meaning here.